src/components/App/index.tsx

```diff
@@ -152,14 +156,39 @@ export class AppBase extends React.Component<Props> {
     );
   }
 
+  renderWithLinterProvider = ({ messageMap }: LinterProviderInfo) => {
+    return messageMap ? (
+      <div className={styles.linterMessagesShell}>
+        {messageMap.general.map((message) => {
+          return (
+            <LinterMessage
+              className={styles.linterMessage}
+              key={message.uid}
+              message={message}
+            />
+          );
+        })}
+      </div>
+    ) : null;
+  };
+
   render() {
-    const { loading } = this.props;
+    const { currentVersion, loading } = this.props;
 
     return (
       <FullscreenGrid>
         <Header>
           {!loading && <Navbar />}
           {this.renderErrors()}
+          {currentVersion && (
+            <LinterProvider
+              versionId={currentVersion.id}
+              validationURL={currentVersion.validationURL}
+              selectedPath={currentVersion.selectedPath}
+            >
+              {this.renderWithLinterProvider}
+            </LinterProvider>
+          )}
```

Comment:
> So, are we okay with leaking version stuff into the `App` component? I'd rather see this in the `Compare` and `Browse` page components or maybe into the `VersionFileViewer`.

I don't quite understand what you mean by leaking -- can you elaborate? In both the current patch (using `App`) and your proposed patch using `VersionFileViewer`, the data is mapped from a Redux store so what's the difference? I don't see what moving the mapping to `VersionFileViewer` solves. Did you mean to use `props.version` from `VersionFileViewer` instead?

Maybe it's because I don't consider the `App` component generic. It's a component specific to the shell of the code manager application. The main piece of data for all pages of code manager is the *current version* (as well as *current user*).

Reply:
> I don't quite understand what you mean by leaking -- can you elaborate? 

> Maybe it's because I don't consider the `App` component generic. It's a component specific to the shell of the code manager application. The main piece of data for all pages of code manager is the _current version_ (as well as _current user_).

> I don't see what moving the mapping to VersionFileViewer solves. 

Sorry. I mean putting code where it isn't supposed to be. The `App` component is the shell of all the page-level components. While having the current version in absolutely all pages _might_ make sense, I don't think having linter messages in all the pages does.

Example: in the near future, we might want to expose results of our various secret tools. Such results would (probably) be tied to a version and would likely be displayed on their own pages. We don't want to have (general) linter messages on these pages.

Reply:
ok, I see what you mean. That is a good reason for moving the logic to `VesionFileViewer` (the shared component for all version viewing). I will try moving it there. However, we probably won't get this separation 100% right because it's hard to predict the future.